<img src="../../images/Callysto_Notebook-Banner_Top_06.06.18.jpg"/>

In [1]:
%%html

<script>
  function code_toggle() {
    if (code_shown){
      $('div.input').hide('500');
      $('#toggleButton').val('Show Code')
    } else {
      $('div.input').show('500');
      $('#toggleButton').val('Hide Code')
    }
    code_shown = !code_shown
  }
  
  $( document ).ready(function(){
    code_shown=false;
    $('div.input').hide()
  });
</script>
<form action="javascript:code_toggle()"><input type="submit" id="toggleButton" value="Show Code"></form>

# Canada's Residential Schools

The Truth and Reconciliation Commission (TRC) was formally created in the year 2008 in order to bring out to light the experiences of Indigenous students in the residential schools. In December 2015, TRC came up with the final report which gave in detail the history of residential schools, their devastating legacy and the abuse left by these residential schools. The intention of this notebook is to bring awareness about the residential schools and to know about the past associated with this country which will help in understanding why it is important to be supportive and inclusive of different groups of people, launguages and cultures.

## What is a residential school?

![Study time in residential school](images/study-time.jpg)
<center>Study time in a residential school.</center>
<center>Source: Library and Archives Canada, PA-042133</center>

The Canadian government in the 19th century decided that it was responsible for providing education and care for the aboriginal people of the country in order to assimilate them into the Canadian society. The government believed that the only way to achieve this is by separating the students from their families and denying access to their own culture, language, and traditions. The government also believed that the best chance of success for these students was to learn English and to adopt Christianity and Canadian customs. This way the students will pass on the newly adopted lifestyle, language and culture to their children which would completely abolish the native traditions in a few generations.

In order to put these into action, the Canadian government developed a policy called "aggressive assimilation" to be taught at the government-funded educational institutions which were church-run and provided the Indigenous children with rudimentary training in the trades and agriculture. The government believed that the children were easier to mold than adults and the concept of boarding school with compulsory attendance where children were separated from their families and communities was the best way to prepare them for life in the mainstream society.

## Students at the residential school

It is estimated that about 150,000 First Nation, Inuit, and Métis children attended these residential schools. 

**Métis:** Métis Nation's political organization, The Métis National Council (MNC), defines “Metis” as: “a person who self-identifies as Métis, is distinct from other Aboriginal Peoples, is of historic Métis Nation ancestry and who is accepted by the Métis Nation.” Métis people are a mix of European and Indigenous ancestry who trace their origin to Red River valley and the prairies including Ontario, British Columbia, the Northwest Territories and the northern United States.

**Inuit:** Inhabitants of North American Arctic, covering a distance of 6,000 kilometres from the Bering Strait to East Greenland were termed as Inuit, which simply means "the people". Inuit also live in northern Alaska, Greenland and some even in Russia who have culturally common heritage and a common language. The term "Eskimo" was used to identify the Inuit people until recently.

**First Nations:** A term used to refer to about 1.3 million Aboriginal people in Canada who are not Métis or Inuit and who are the original inhabitants of the land that is now called Canada. There are [634 First nations governments or band ](https://en.wikipedia.org/wiki/List_of_First_Nations_governments) in Canada, who speak more than 50 different languages.

### Questions

1. Identify two recognized First nations governments in your province.

## Issues at residential schools

### Quality of Education:

The quality of education and skills that were taught at the residential schools were generally substandard. The students were often trained for menial labor positions that many were not able to adapt or function in an urban setting.

### Language, Values, beliefs, culture and traditions:

The schools were increasingly run by well-established churches in Canada, which meant the students were exposed to Christian values, beliefs, and traditions. The students were not allowed to wear traditional clothing or accessories. They were given uniform clothing (mostly of substandard quality), a number assigned to each student and / or a Christian name. Children were punished if they were found using their traditional names, which further separated the children from their culture and idenitities.

Students were encouraged to speak in either English or French and were restrained from speaking their first language or even practice any native traditions. The students were belittled and punished when they were found speaking in their first language or practicing their traditions. As a result of this students were deprived of speaking and learning their first language and made them feel like they didn't belong in their own family whenever they went back to their reserves and made them feel ashamed of their native heritage. Without the knowledge of their first language comes the loss of the ability to understand the teachings and stories that were passed down in Indigenous cultures that value oral tradition.

### Separation from family:

Attendance was initially voluntary but later the government became more stringent and attendance was made compulsory. The idea was to separate the children as far away as possible from their family and communities to sever connections to their roots. Students never got to experience the normal life outside these schools as most students in these residential schools spent almost 10months of a year in the school, away from their parents; some even stayed all year round. Students were also forced to have all correspondence sent to their parents be written in English, which many parents couldn't read, which further broke ties with their families.

### Poor nutrition:

Dr. Peter Bryce reported in 1907, medical inspector for the Department of Indian affairs, found that the students were not given nutritious food, the living conditions were often very poor and the schools were overcrowded. It was also noted that the school buildings were often poorly constructed with bad ventilation and poor heating which paved way for various disease outbreaks.

### Abuse:

Students were often subject to emotional, physical and sexual abuse. Since the schools were overcrowded the students did not receive any personal attention and the lack of love or care from family and the fear of being punished for making mistakes contributed to long-term depression, anxiety, and disconnect. About half of the 80,000 survivors of the residential school system reported of being seriously abused either physically or sexually. These cases went unreported or often ignored by the institutions in order to save their own face than protecting the children.

### Missing or gone:

Many students tried to escape these residential schools, the harsh punishments and abuse to be back with their parents often risking their own lives without supplies, help or proper clothing. Children were often caught by officials and brought back to the schools to be severely punished. A number of children even lost their lives trying to escape the horrible conditions to return to their loved ones.

### Questions

1. Can you relate a time where a relative passed on a particular skill to you?
2. Can you relate a story that has been passed down to you from an older generation that you would be able to tell the next generation? What part(s) of the story offer a learning aspect?

### Activity

Below is a video interview of one of the survivors of the residential school system. Watch the video carefully and answer the questions that follow.

Video source: http://www.cbc.ca/player/play/1178497603731/

In [2]:
from IPython.display import HTML
HTML('<iframe src="//www.cbc.ca/i/caffeine/syndicate/?mediaId=1178497603731" width="640" height="360" frameborder="0" allowfullscreen></iframe>')

/opt/conda/lib/python3.6/site-packages/IPython/core/display.py:689: UserWarning: Consider using IPython.display.IFrame instead
  warnings.warn("Consider using IPython.display.IFrame instead")


### Questions

1. What key issues did Wanbdi discuss in the video? ( Choose all that apply: Anger, physical abuse, flight, poor nutrition, racism, pride, mental abuse)

2. What consequences did Wanbdi face for running away from residential school?

3. What are his recommendations to Canadians on reconciliation?

## Geographic locations of residential schools

![Mohawk School](images/mohawk-school.jpg)
***The first residential school - Mohawk Institute Residential School in Ontario (Library and Archives Canada)***

In order to understand and acknowledge the truth behind these devastating residential schools, it is important to know the locations of such schools. From this, we can understand how and where it all began, which was the first residential school and how the idea quickly progressed to other provinces. Knowing the history of the place you live is important to give the respect that the place and people that belong to that place deserve and also critical to form meaningful relationships with them.

To begin with we first download open data set available at http://hesperus-wild.org/GIS_carto/IRS.html that contains the locations and other information about the residential schools.

In [3]:
from fastkml import  kml
from lxml import etree
import pandas as pd
import pandas as pd
import matplotlib.pyplot as plt
from plotly.offline import init_notebook_mode, iplot
import plotly.graph_objs as go
import plotly.plotly as py
init_notebook_mode(connected=True)

kml_file = 'data/IRS.kml'
myfile = open(kml_file, 'rt', encoding='utf-8')
doc=myfile.read().encode('utf-8')

k = kml.KML()
k.from_string(doc)

for doc in k.features():
    for folder in doc.features():
        value_dict = []
        for plcm in folder.features():
            
            for schema_data in plcm.extended_data.elements:
                
                data_dict ={}
                for data in schema_data.data:
                    
                    data_dict[data['name']] = data['value']               
            value_dict.append(data_dict)                       

residential_df = pd.DataFrame(value_dict)
residential_df['Longitude'] = pd.to_numeric(residential_df['Longitude'],errors='ignore')
residential_df['Latitude'] = pd.to_numeric(residential_df['Latitude'],errors='ignore')
residential_df



Atlas_pg                                         Descriptio  From  \
0          1            Chateh, AB: Lot 1, north of Sousa creek  1951   
1          2  Present locations of the Onchimnahoos School i...  1898   
2          3  Present location of Blue Quills First Nation's...  1931   
3          4  At Blackfoot, Crossing, on the north side of t...  1900   
4          5  just west of Cluny where Range Road 214A meets...  1909   
5          6  In the old Wabasca Settlement,at what is today...  1902   
6          7  current location of the Poundmaker's Lodge Tre...  1824   
7          8  Maskwasis, AB, across the street from the Ermi...  1895   
8          9            On River Road, just west of 50th Street  1903   
9         10        Site of North Lakes College, Grouard Campus  1894   
10        11                 Near Mission Point, today's Lot 2A  1874   
11        12  Just west of Joussard, AB, where Mission Creek...  1913   
12        13     Lac La Biche Mission, 16 km NW of Lac La Biche  1893   
13        14  Lot 64, SEC 13, T76 R15: 300m west of the Kapa...  1895   
14        15  McDougal Memorial wayside on the Bow Valley Tr...  1885   
15        16        in the area of the Nakoda Elementary School  1922   
16        17              Approx. 5km southwest of Gleichen, AB  1886   
17        18              Approx. 4.5 km southwest of Cluny, AB  1889   
18        19  At the site of Old Sun Community College, Lot ...  1912   
19        20  About 8 km due east of Brocket, in SW ¼ SEC 7 ...  1887   
20        21                       Mission Park, St. Albert, AB  1873   
21        22  next to the Peace River Correctional Centre, o...  1900   
22        23  Victoria Mission Historic Site, 3.5km west of ...  1890   
23        24         SEC D T6 R28, 4.5 km southeast of Brocket.  1927   
24        25  On the west bank of the High River, 1.8km SW o...  1884   
25        26                                    In Standoff, AB  1911   
26        27                         Red Crow Community College  1926   
27        28  on the west side of the Belly River and within...  1893   
28        29               At Ninastoko, 7km NW of Cardston, AB  1925   
29        30          Old Agency Road, 2.8 km south of Two Guns  1892   
..       ...                                                ...   ...   
134      135                 within the community of Mistassini  1971   
135      136  On the northeast side of the confluence of the...  1895   
136      137                         5.3km southeast of Kamsack  1888   
137      138          500m southwest of the Okanese Band Office  1889   
138      139              one half mile east of St. Philips, SK  1905   
139      140  1100m southwest of the George Gordon First Nat...  1888   
140      141  on what is now the Wa-Pii Moos-toosis (White C...  1884   
141      142  site of Cowessess Community Education Centre n...  1936   
142      143                 NW ¼ SEC 14, Township 27, Range 15  1932   
143      144  Former army barracks just east of Victoria Hos...  1944   
144      145                    near the La Ronge health centre  1906   
145      146                          1.6 km west of Onion Lake  1893   
146      147  In Section 11, Township 54, Range 27, 1.75 km ...  1926   
147      148       22nd St. between 1st Ave. W and Central Ave.  1948   
148      149  west end of Ritter Avenue, now the Pual Dojack...  1890   
149      150  On the north side of the Qu'Appele River where...  1885   
150      151                          1.6 km west of Onion Lake  1894   
151      152                    Section 8, Township 54 Range 27  1927   
152      153               one half mile south of Duck Lake, SK  1894   
153      154  In the hamlet of St. Philips, SK, 500m south o...  1928   
154      155        on the west bank of the Sturgeon Weir River  1926   
155      156                 north of 99th Street in Delmas, SK  1901   
156      157               In the settlement of Badgerville, S

Now that we have the data on hand, lets try to plot them on a graph to  find the locations of the schools. 

In [4]:
# residential_df.plot(kind="scatter", x="Longitude", y="Latitude", alpha=0.4, label="Residential School",
#         figsize=(10,7),)
# plt.show()

trace = go.Scatter(
    y = residential_df['Latitude'],
    x = residential_df['Longitude'],
    text = residential_df['map_name']+ '<br>'+'From:  '+residential_df['From']+'<br>' +'To:  '+residential_df['To'],
    mode='markers',
    marker = dict(
    opacity = 0.5,
    color = 'blue')
    )

layout = go.Layout(
    title='Latitude vs Longitude - Scatter plot',
    xaxis=dict(
        title='Longitude'
    ),
    yaxis=dict(
        title='Latitude'
    )
)

data = [trace]

fig = dict(data=data, layout=layout)
iplot(fig, filename='styled-scatter')

What we see above is a scatter plot to diplay two values of our dataset Latitude vs Longitude, but what do they mean and how are we supposed to use it? To get a better understanding of Latitudes and Longitudes, let us take a quick detour to explore a little more about these concepts. Before we do that lets test your knowledge about scatter plots. Try and answer the questions that follow.

### Questions

1. What is a scatter plot?
2. What two variables are related in the scatterplot? 
3. What do each dot on the plot represent?
4. Why are some dots lighter and some darker? (Hint: Zoom in on darker dots)


## Brief Detour About Geography

**Latitudes and longitudes** are very basic terms in geography. They help us to explain the location or position of something anywhere in the world. Latitudes and longitudes form a grid system on the globe, and by knowing the shape of this grid, we can easily locate any place on Earth.

### Latitude

Lines of latitudes are horizontal systems of measurement that encircle the globe from east to west. Latitudes are sometimes called "parallels," as they are at an equal distance from each other and never meet.

Latitude lines measure how far north or south of the equator something is. The equator is defined as the starting point for measuring latitude - that's why it is marked as 0 degrees latitude. The North Pole is listed at 90 degrees, and the South Pole is listed at -90 degrees. The more "degrees latitude" a location is, the further away it is from the equator. For example, 50 degrees latitude is further from the equator than 20 degrees latitude is. Depending on if a location is above or below the equator, they are listed as "degrees North" or "degrees South", and are given a matching positive (+) or negative (-) symbol. 


Each degree of latitude is approximately 111 kilometers apart (69 miles), although there is some variation as the Earth is not perfectly round. The globe below has the parallels drawn on it. Feel free to spin the globe and zoom in or out on any location. Parallels located south of the Equator are orange, and this part of the Earth is called the Southern Hemisphere. Conversely, parallels located north to the Equator are blue, and this part of the Earth is called the Northern Hemisphere.


In [5]:

contours = []

equator_lon = [-180,-170,-160,-150,-140,-130,-120,-110,-100,-90,-80,-70,-60,-50,-40,-30,-20,-10,0,
               10,20,30,40,50,60,70,80,90,100,110,120,130,140,150,160,170,180]
euator_lat = [0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0]

#equator
contours.append( dict(
        type = 'scattergeo',
        hoverinfo='text',
        text= '0° equator',
        name = '0° equator',
        lon = equator_lon,
        lat = euator_lat,
        mode = 'lines',
        line = dict(
            width = 3,
            color = 'green'
        )
    ) )
#north pole
contours.append( dict(
        type = 'scattergeo',
        hoverinfo='text',
        text= '90° north pole',
        name = '90° north pole',
        lon = [0],
        lat = [90],
        mode = 'markers',
        line = dict(
            width = 2,
            color = 'rgb(50,136,189)'
        )
    ) )
#north hemispere
for i in range (1,9):
    contours.append( dict(
        type = 'scattergeo',
        hoverinfo='text',
        text= str(10*i)+'° north',
        name = str(10*i)+'° north',
        lon = equator_lon,
        lat = [x+10*i for x in euator_lat],
        mode = 'lines',
        line = dict(
            width = 2,
            color = 'rgb(50,136,189)'
        )
    ) )
    
#south hemispere
for i in range (1,9):
    contours.append( dict(
        type = 'scattergeo',
        hoverinfo='text',
        text= str(-10*i)+'° south',
        name = str(-10*i)+'° south',
        lon = equator_lon,
        lat = [x-10*i for x in euator_lat],
        mode = 'lines',
        line = dict(
            width = 2,
            color = 'rgb(244,109,67)'
        )
    ) )
    
contours.append( dict(
        type = 'scattergeo',
        hoverinfo='text',
        text= '-90° south pole',
        name = '-90° south pole',
        lon = [0],
        lat = [-90],
        mode = 'markers',
        line = dict(
            width = 2,
            color = 'rgb(244,109,67)'
        )
    ) )

layout = dict(
        title = 'Parallels over globe<br>(Click and drag to rotate)',        
        geo = dict(
            showland = True,
            showlakes = True,
            showcountries = True,
            showocean = True,
            countrywidth = 0.5,
            lakecolor = 'rgb(255, 255, 255)',
            oceancolor = 'rgb(255, 255, 255)',
            landcolor = 'rgb(212, 212, 212)',
            projection = dict( 
                type = 'orthographic',
                rotation = dict(
                    lon = -100,
                    lat = 40,
                    roll = 0
                )            
            ),

        )
    )
    
fig = dict( data=contours, layout=layout )
iplot( fig, validate=False )

### Questions

Look at the globe. Try rotating it and hover your mouse over different parallels. 
Every tenth parallel is displayed.
If you click on any parallel listing on the right, the latitude line will either appear or disappear on the map.

1. Can you find the Equator? Which continents does it go through?
2. Find Canada. Is it located in the Southern or Northern Hemisphere?
3. Which latitude lines is Canada located between?

### Longitude
In contrast to latitude, longitude lines Lines are vertical systems of measurement that encircle the globe from north to south. Longitude lines converge at poles and are widest at the equator. Longitude lines can also be called "meridian lines". Longitude shows how far a location is east or west of a universal vertical line, called the Prime Meridian. The Prime Meridian line runs through the British Royal Observatory in Greenwich England, from the North Pole to the South Pole. As the vertical starting point for longitude, the Prime Meridian is defined as 0 degrees longitude. Interestingly, for many years, each country had its own prime meridian. In 1884, a group of scientists, navigators, and business owners made Greenwich, England, the official worldwide Prime Meridian.

Longitude lines span 180 degrees both east and west of the Prime Meridian. Longitude lines that are **east** of the prime meridian are positive (+), and lines that are **west** of the Prime Meridian are negative (-). The longitude line that is exactly 180 degrees east and west of the Prime Meridian is known as the International Date Line (it is directly opposite to the Prime Meridian). 

Lines of longitude at every ten degrees are shown on the globe below, try rotating it and hover your mouse over different meridians. If you click on any meridian listed on the right it will disappear/reappear on the map. On this map, the blue meridians located east of the Prime Meridian are part of the Eastern Hemisphere. The orange meridians located west to the Prime Meridian are part of the Western Hemisphere.

In [6]:
contours = []

mer_lat1 = [90,100,110,120,130,140,150,160,170,180,-180,-170,-160,-150,-140,-130,-120,-110,-100,-90]
mer_lon1 = [0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0]

mer_lat2 = [-90,-80,-70,-60,-50,-40,-30,-20,-10,0,10,20,30,40,50,60,70,80,90]
mer_lon2 = [0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0]

#zero meridian
contours.append( dict(
        type = 'scattergeo',
        hoverinfo='text',
        text= '0° Prime meridian',
        name = '0° Prime meridian',
        lon = mer_lon2,
        lat = mer_lat2,
        mode = 'lines',
        line = dict(
            width = 3,
            color = 'green'
        )
    ) )

contours.append( dict(
        type = 'scattergeo',
        hoverinfo='text',
        text= '180° International Date Line',
        name = '180° International Date Line',
        lon = mer_lon1,
        lat = mer_lat1,
        mode = 'lines',
        line = dict(
            width = 3,
            color = 'green'
        )
    ) )

#west hemispere
for i in range (1,18):
    contours.append( dict(
        type = 'scattergeo',
        hoverinfo='text',
        text= str(10*i-180)+'° west',
        name = str(10*i-180)+'° west',
        lat = mer_lat1,
        lon = [x+10*i for x in mer_lon1],
        mode = 'lines',
        line = dict(
            width = 2,
            color = 'rgb(244,109,67)'
        )
    ) )
    
#east hemispere
for i in range (1,18):
    contours.append( dict(
        type = 'scattergeo',
        hoverinfo='text',
        text= str(10*i)+'° east',
        name = str(10*i)+'° east',
        lat = mer_lat2,
        lon = [x+10*i for x in mer_lon2],
        mode = 'lines',
        line = dict(
            width = 2,
            color = 'rgb(50,136,189)'
        )
    ) )
    


layout = dict(
        title = 'Meridians over globe<br>(Click and drag to rotate)',
       # showlegend = False,         
        geo = dict(
            showland = True,
            showlakes = True,
            showcountries = True,
            showocean = True,
            countrywidth = 0.5,
            lakecolor = 'rgb(255, 255, 255)',
            oceancolor = 'rgb(255, 255, 255)',
            landcolor = 'rgb(212, 212, 212)',
            projection = dict( 
                type = 'orthographic',
                rotation = dict(
                    lon = -100,
                    lat = 40,
                    roll = 0
                )            
            ),
    
        )
    )
    
fig = dict( data=contours, layout=layout )
iplot( fig, validate=False )

### Questions



1. Can you locate the Prime Meridian? 
    A. What city in Great Britain does it pass through? 
    B. What other countries does the Prime Meridian pass through?
2. Locate Canada on the globe. Is Canada located in the Eastern or Western Hemisphere?
3. Approximately what lines of latitude is Canada located between?

## Looking at Canada's map
Lets display some of the Canadian cities on the map.

From the map below, if you hover your mouse over different cities, their latitude and longitude locations are  displayed as a pair of numbers: `(longitude, latitude)`. For example, if you hover your mouse over Ottawa, you will see: (-75.7, 45.42). This tells us that that Ottawa is located 75.7 degrees west (longitude) and 45.42 degrees north (latitude).

In [7]:
import plotly.plotly as py
import pandas as pd

df = pd.read_csv('https://raw.githubusercontent.com/plotly/datasets/master/2011_february_us_airport_traffic.csv')
df.head()
name=["Edmonton", "Victoria","Winnipeg","Fredericton","St. John's","Halifax","Toronto",
      #"Charlottetown","Quebec City", 
      "Regina","Whitehorse","Iqaluit","Yellowknife", "Ottawa"]
long=[-113.49,-123.37,-97.14,-66.64, -52.71, -63.57, -79.38,
      #-63.13,-71.21,
      -104.62,-135.06,-68.52,-114.38,-75.70]
latt=[53.54,48.43,49.90,45.97, 47.56, 44.65, 43.65,
      #46.238240,46.813878,
      50.44,60.72,63.75,62.45,45.42]
df['text'] = df['airport'] + '' + df['city'] + ', ' + df['state'] + '' + 'Arrivals: ' + df['cnt'].astype(str)

data = [ dict(
        type = 'scattergeo',
        locationmode = 'north america',
        lon = long,
        lat = latt,
        text = name,
        mode = 'markers+text',
        textposition= 'top',
        marker = dict(
            size = 8,
            opacity = 0.8,
            reversescale = True,
            autocolorscale = False,
            symbol = 'square',
            line = dict(
                width=1,
                color='rgba(102, 102, 102)'
            ),
            color = 'red',
        ))]

layout = dict(
        title = "Some Canada's provincial capitals <br>(Hover for coordinates)",
        colorbar = True,
        geo = dict(
            scope = 'north america',
            showland = True,
            landcolor = "rgb(212, 212, 212)",
            countrycolor = "rgb(255, 255, 255)",
            showlakes = True,
            lakecolor = "rgb(255, 255, 255)",
            showsubunits = False,
            showcountries = True,
            resolution = 50,
            projection = dict(
                type = 'kavrayskiy7',
            ),
            lonaxis = dict(
                showgrid = True,
                gridcolor ='rgb(244,109,67)',
                gridwidth = 2,
                range= [ -140, -50 ],
                dtick = 10
            ),
            lataxis = dict (
                showgrid = True,
                gridcolor ='rgb(50,136,189)',
                gridwidth = 2,
                range= [ 40.0, 80.0 ],
                dtick = 10
            )
        )
    )

fig = dict( data=data, layout=layout )
iplot( fig, validate=False )

### Questions 

1. Are there any negative values of latitude and longitude in Canada? Why or why not?
2. What's the northern most Canadian city on the map? What are its coordinates?
3. Locate Winnipeg and Regina, and note down their coordinates. What do you notice about their latitude values? What does this tell you?
4. Locate both Edmonton and Yellowknife.
    A. What is the difference between their latitudes?
    B. What does the distance between their latitudes tell you?
    C. Given that 1 degree of latitude covers approximately 111 km, calculate the distance between these cities. 

(Answer: (62.45-53.54)*111=989.01km )


Now that we have a good understanding of what Latitude and Longitudes are, let's use them to plot the residential schools on a map. 

In [8]:
df = pd.read_csv('https://raw.githubusercontent.com/plotly/datasets/master/2011_february_us_airport_traffic.csv')
df.head()
name=["Edmonton", "Victoria","Winnipeg","Fredericton","St. John's","Halifax","Toronto",
      #"Charlottetown","Quebec City", 
      "Regina","Whitehorse","Iqaluit","Yellowknife", "Ottawa"]
long=[-113.49,-123.37,-97.14,-66.64, -52.71, -63.57, -79.38,
      #-63.13,-71.21,
      -104.62,-135.06,-68.52,-114.38,-75.70]
latt=[53.54,48.43,49.90,45.97, 47.56, 44.65, 43.65,
      #46.238240,46.813878,
      50.44,60.72,63.75,62.45,45.42]
df['text'] = df['airport'] + '' + df['city'] + ', ' + df['state'] + '' + 'Arrivals: ' + df['cnt'].astype(str)


N = 500

trace0 = go.Scattergeo(
    locationmode = 'country names',
    lon = long,
    lat = latt,
    text = name,
    name = 'Cities',
    mode = 'markers+text',
        textposition= 'top right',
        marker = dict(
            size = 8,
            opacity = 0.8,
            reversescale = True,
            autocolorscale = False,
            symbol = 'square',
            
            color = 'red',
        )
)

trace1 = go.Scattergeo(
    lat = residential_df['Latitude'],
    lon = residential_df['Longitude'],
    text = residential_df['map_name']+ '<br>'+'From:  '+residential_df['From']+'<br>' +'To:  '+residential_df['To'],
    name = 'Schools',
    marker = dict(
    opacity = 0.8,
    color = 'blue')
    )

data = [trace0, trace1]

layout = dict(
        geo = dict(
            scope = 'north america',
            showland = True,
            landcolor = "rgb(212, 212, 212)",
            countrycolor = "rgb(255, 255, 255)",
            showlakes = True,
            lakecolor = "rgb(255, 255, 255)",
            showsubunits = True,
            showcountries = True,
            resolution = 50,
            projection = dict(
                type = 'kavrayskiy7',
            ),
            lonaxis = dict(
                showgrid = False,
                range= [ -140, -50 ],
                dtick = 10
            ),
            lataxis = dict (
                showgrid = False,
                range= [ 40.0, 80.0 ],
                dtick = 10
            )
        )
             )

fig = dict(data=data, layout=layout)
iplot(fig, filename='styled-scatter')

### Questions



1. What was the last residential school to be opened and in which province was it located?
(Hint: Hover over each marker to find out the name and year of operation)
2. Locate a residential school that is close to where you currently live or go to school. Write down the name of the school and years of operation. 
3. How many schools were there in your province of residence?

To make answering to question 3 in the above section a little easier, let us build a table by grouping number of schools by their province.

In [9]:
province_count = residential_df.groupby(by='Province o', as_index=False).agg({'Atlas_pg': pd.Series.nunique})
province_count.columns = ['Province', 'No. of Schools']
province_count.sort_values(by=['No. of Schools'])

Province  No. of Schools
4            Nova Scotia               1
9                  Yukon               6
7                 Québec              12
5                Nunavut              13
3  Northwest Territories              15
2               Manitoba              17
1       British Columbia              21
6                Ontario              23
8           Saskatchewan              23
0                Alberta              33

The resulting table shows that the province with least residential schools was Nova Scotia with only 1 school in the provicne and Alberta had most number of residential schools with about 33 schools in the province. 

The Shingwauk Residential School Centre at Algoma University has put together a dataset with list of residential schools with information about their names, location, dates of operation, religion that the school followed, and some notes about them. It is to be noted that this may not be a definitive or complete list as it was created as part of their work in process as a community archive.

Take a look at this second dataset below and try to answer the questions that follow.

In [10]:
shingwauk_df = pd.read_excel('data/IRSList2018.xlsx')
shingwauk_df

School Name  \
0              Ahousaht Indian Residential School   
1                                   Akaitcho Hall   
2      Aklavik Anglican Indian Residential School   
3     Aklavik Catholic Indian Residential School    
4        Albany Mission Indian Residential School   
5               Alberni Indian residential School   
6           Alexandra Industrial School for Girls   
7                    All Hallows School for Girls   
8            All Saints Indian Residential School   
9            All Saints Indian Residential School   
10                      Alnwick Industrial School   
11                 Amos Indian Residential school   
12            Armstrong Indian Residential School   
13                      Aroland Indian Day School   
14           Assiniboia Indian Residential School   
15           Assumption Indian Residential School   
16              Baptist Indian residential School   
17                   Battleford Industrial School   
18              Beauval Indian Residential School   
19                   Big Trout Lake Indian School   
20               Birtle Indian Residential School   
21                  Bishop Horden Memorial School   
22          Blue Quills Indian Residential School   
23          Bompas Hall Indian Residential School   
24                      Brandon Industrial School   
25                                 Breynant Hall    
26          Camperville Indian Residential School   
27             Carcross Indian Residential School   
28             Carcross Indian Residential School   
29                  Caughnawaga Indian Day School   
..                                            ...   
142      St. Margaret's Indian Residential School   
143                    St. Martin Boarding School   
144          St. Mary's Indian Residential School   
145          St. Mary's Indian Residential School   
146  St. Mary's Mission Indian Residential School   
147       St. Michael's Indian Residential School   
148       St. Michael's Indian Residential School   
149                               St. Paul's Hall   
150          St. Paul's Indian Residential School   
151                  St. Paul's Industrial School   
152  st. Paul Des Métis Indian Residential School   
153                       St. Peter Claver School   
154         St. Peter's Indian Residential School   
155         St. Peter's Indian Residential School   
156         St. Philips Indian Residential School   
157                                 Stringer Hall   
158         Sturgeon Lake Indian residential Lake   
159           St. Rupert's Land Industrial School   
160       Thomas Crosby Indian Residential School   
161        Thunderchild Indian Residential School   
162              Victoria Catholic Convent School   
163                         Washakada Indian Home   
164            Waterhen Indian Residential School   
165            Wawanosh Indian Residential School   
166         Wikwemikong Indian Residential School   
167       Williams Lake Indian Residential School   
168                Yale Indian Residential School   
169            Youville Indian Residential School   
170                                    Yukon Hall   
171              Yuquot Indian Residential School   

                             Location      Dates        Religion  \
0                     Ashousaht, B.C.  1901-1950    Presbyterian   
1                 Yellowknife, N.W.T.  1958-1994         Federal   
2    Single Point and Aklavik, N.W.T.  1924-1959        Anglican   
3                     Aklavik, N.W.T.  1925-1930  Roman Catholic   
4                    Fort Albany, Ont  1912-1963  Roman Catholic   
5                Prince Albert, Sask.  1920-1973   United Church   
6                    Toronto, Ontario     1897-?             NaN   
7                            Yale, BC  1884-1914        Anglican   
8                        Aklavik, NWT  1936-1959        Anglican   
9                    Lac La Ronge, SK  1906-1947        Anglican   
10 

### Questions

1. Can you identify two differences between this dataset and the previous one? Please describe them.
2. What information is missing that is required to plot these schools on a map?

Like we already discussed in previous sections, to plot a location on map we need to have two data points, latitude and longitude. Geocoding the process of converting addresses into geographic coordinates (latitudes and longitudes). Below you'll see the complete geocoded dataset with the latitude and longitude values pointing to the city mentioned under the location column.

In [11]:
### This is the code that was used to geocode the locations and write them to a new file

# # we may have to download the geocoder before running the rest of the code.
# #!pip install -U geocoder
# import geocoder
# import time
# import csv

# # Read the file and get Residential schools into a list.
# file_in = pd.read_excel('data/IRSList2018.xlsx', sheet_name=0, header=0)

# # Get the location information into a list
# addresses = file_in['Location']

# # Loop over the location list and geocode each location and add their coordinates to a list
# results = []
# geocode_done = False
# for address in addresses:
#     geocode_result = geocoder.google(address)
#     print(geocode_result.status)
#     if geocode_result.status == 'OVER_QUERY_LIMIT':
#         time.sleep(120)
#         geocode_result = geocoder.google(address)
#     elif geocode_result.status != 'OK':
#         geocode_result = geocoder.osm(address)
    
#     if geocode_result.status == 'ERROR - No results':
#             time.sleep(60)
#             geocode_result = geocoder.google(address)
        
#     results.append(geocode_result.json)
# results

# # Loop over the latitude and longitude list and add them to the 
# # original pandas dataframe that has the rest of the school information

# lat_list = []
# lng_list = []
# for result in results:
#     if result is not None:
#         lat_list.append(result['lat'])
#         lng_list.append(result['lng'])
#     else:
#         print("its none")
#         lat_list.append(0)
#         lng_list.append(0)
        
        
# file_in["latitude"] = lat_list
# file_in["longitude"] = lng_list

# # write the dataframe to a file
# file_in.to_csv("data/IRSlist_with_latlng.csv")

In [12]:
latlng_df = pd.read_csv('data/IRSlist_with_latlng.csv', sep="\t")
latlng_df.loc[latlng_df["Dates"].isnull()]

School Name                 Location Dates  \
13            Aroland Indian Day School                  Ontario   NaN   
19         Big Trout Lake Indian School     Big Trout Lake, Ont.   NaN   
25                       Breynant Hall        Fort Smith, N.W.T.   NaN   
29        Caughnawaga Indian Day School        Caughnawaga, Ont.   NaN   
35           Christ the King Day School       Moosonee, Ontario.   NaN   
49       Federal Hostel at George River    Kangirsualujjuaq, QC.   NaN   
50  Federal Hostel at Great Whale River        Kuujjuarapik, QC.   NaN   
51          Federal Hostel at Payne Bay  Kangirsuk (Bellin), QC.   NaN   
52      Federal Hostel at Port Harrison            Inukjuak, QC.   NaN   
81                  Lac Seul Day School           Lac Seul, Ont.   NaN   
82                  Long Lac Day School           Long Lac, Ont.   NaN   
98          Northwestern Bay Day School       Fort Frances, Ont.   NaN   

          Religion Notes   latitude   longitude  
13             NaN   NaN  51.253775  -85.323214  
19             NaN   NaN  53.758588  -90.076390  
25  Roman Catholic   NaN  60.005520 -111.884938  
29  Roman Catholic   NaN  45.410785  -73.681476  
35             NaN   NaN  51.273089  -80.640049  
49         Federal   NaN  58.710155  -65.995116  
50         Federal   NaN  55.274570  -77.763784  
51         Federal   NaN  60.023415  -70.026235  
52         Federal   NaN  58.455121  -78.105103  
81             NaN   NaN  50.332085  -92.481994  
82             NaN   NaN  49.781666  -86.534706  
98             NaN   NaN  48.609949  -93.395528

Now let us plot the schools onto the previous map, so we can compare and contrast the two datasets. The blue markers represent the residential schools listed in the first dataset while yellow markers respresent the schools listed in the second dataset. 

Since we geolocated the latitude and longitude values in the second dataset to their nearest city or town listed in the location column, they may not represent the precise location of the school while in the first dataset we had the precise geographic coordinates for most of the schools. If you hover over the data points on the map you'll see that some of schools from the two datasets align perfectly at the same location while some do not. 

The orange or brown markers are nothing but a few of blue and yellow markers super imposed at a single location.

In [13]:
df = pd.read_csv('https://raw.githubusercontent.com/plotly/datasets/master/2011_february_us_airport_traffic.csv')
df.head()
name=["Edmonton", "Victoria","Winnipeg","Fredericton","St. John's","Halifax","Toronto",
      #"Charlottetown","Quebec City", 
      "Regina","Whitehorse","Iqaluit","Yellowknife", "Ottawa"]
long=[-113.49,-123.37,-97.14,-66.64, -52.71, -63.57, -79.38,
      #-63.13,-71.21,
      -104.62,-135.06,-68.52,-114.38,-75.70]
latt=[53.54,48.43,49.90,45.97, 47.56, 44.65, 43.65,
      #46.238240,46.813878,
      50.44,60.72,63.75,62.45,45.42]
df['text'] = df['airport'] + '' + df['city'] + ', ' + df['state'] + '' + 'Arrivals: ' + df['cnt'].astype(str)


N = 500

trace0 = go.Scattergeo(
    locationmode = 'country names',
    lon = long,
    lat = latt,
    text = name,
    name = 'Cities',
    mode = 'markers+text',
        textposition= 'top right',
        marker = dict(
            size = 8,
            opacity = 0.8,
            reversescale = True,
            autocolorscale = False,
            symbol = 'square',
            
            color = 'red',
        )
)

trace1 = go.Scattergeo(
    lat = residential_df['Latitude'],
    lon = residential_df['Longitude'],
    text = residential_df['map_name']+ '<br>'+'From:  '+residential_df['From']+'<br>' +'To:  '+residential_df['To'],
    name = 'Schools dataset 1',
    marker = dict(
    opacity = 0.8,
    color = 'blue')
    )

trace2 = go.Scattergeo(
    lat = latlng_df['latitude'],
    lon = latlng_df['longitude'],
    text = latlng_df['School Name']+ '<br>' + latlng_df['Location']+ '<br>'+'From - To '+ latlng_df['Dates']+'<br>' +':  ',
    name = 'Schools dataset 2',
    marker = dict(
    opacity = 0.6,
    color = 'orange')
    )

data = [trace0, trace1, trace2]

layout = dict(
        geo = dict(
            scope = 'north america',
            showland = True,
            landcolor = "rgb(212, 212, 212)",
            countrycolor = "rgb(255, 255, 255)",
            showlakes = True,
            lakecolor = "rgb(255, 255, 255)",
            showsubunits = True,
            showcountries = True,
            resolution = 50,
            projection = dict(
                type = 'kavrayskiy7',
            ),
            lonaxis = dict(
                showgrid = False,
                range= [ -140, -50 ],
                dtick = 10
            ),
            lataxis = dict (
                showgrid = False,
                range= [ 40.0, 80.0 ],
                dtick = 10
            )
        )
             )

fig = dict(data=data, layout=layout)
iplot(fig, filename='styled-scatter')

### Questions

1. Identify 1 school in your province (if any) that has locations slightly different in the two datasets and note down their geographic coordinates.
2. Identify 1 school in your province (if any) whose location information match pretty closely in the two datasets and note down their geographic coordinates. 
3. Why is it important to have precise location information?

## Calls to action

The report that the Truth and Reconciliation commission published after hearing to thousands of residential school survivors paved way to construct 94 Calls to action: which were individual instructions to guide governments, communities and faith groups down the road to reconciliation. These 94 calls to action can be categorized into six themes: Child Welfare, Education, Language & Culture, Health, Justice, Reconciliation.

CBCNews put up an interactive webpage called [Beyond94](www.cbc.ca/beyond94) dedicated to track the progress of these 94 calls to action. This webpage gives detailed information on all of these 94 calls to action and CBC's analysis on what has been acheived so for, what is in progress and what is yet to be started.

### Questions

Take a look at the [Beyond94](www.cbc.ca/beyond94) page to answer questions below:

1. How many Calls to action are in progress? 
2. List the Calls to actions that has already been completed?
3. Pick any one Call to Action under the theme "Justice", give a summary of what it means and write about its progress.

![alt text](https://github.com/callysto/callysto-sample-notebooks/blob/master/notebooks/images/Callysto_Notebook-Banners_Bottom_06.06.18.jpg?raw=true)